libs/core/src/ports/port.ts

```diff
@@ -0,0 +1,70 @@
+import { ExitCode } from './enums';
+import { Disposable, Disposer } from './interfaces';
+
+/**
+ * Configured adapters map
+ */
+const adapters = new Map<string, Disposable>();
+
+/**
+ * Wraps creation of adapters around factory functions
+ * @param target the factory function
+ * @returns the adapter function
+ */
+export const port =
```

Comment:
What would you think about updating this function to this - assuming there is no logical difference since we don't access `this` context:
```
export function port<T extends Disposable>(target: (arg?: T) => T) {
  return function (arg?: T): T {
    if (!adapters.has(target.name)) {
      const adapter = target(arg);
      adapters.set(target.name, adapter);
      console.log('[binding adapter]', adapter.name || target.name);
    }
    return adapters.get(target.name) as T;
  };
}
```
It's slightly more verbose but reducing the amount of arrow functions makes it easier to follow for me.

Reply:
Done. Thanks for the suggestion!